es6-lib/services/spatial.js

```diff
@@ -1,42 +1,77 @@
-import getDecoder from '../decoders';
+import {
+  getDecoderForExtension
+}
+from '../decoders';
 import Merger from '../decoders/merger';
 import Disk from '../decoders/disk';
+import Core from '../upstream/core';
 import {
-  Core, CoreAuth
+  Auth
 }
-from '../upstream/core';
+from '../upstream/client';
 import async from 'async';
 import _ from 'underscore';
 import * as es from 'event-stream';
 import BBox from '../util/bbox';
 import {
+  parseAMQMessage
+}
+from '../util/hacks';
+import {
   types
 }
 from '../soql/mapper';
 import Layer from '../decoders/layer';
 import config from '../config';
+import logger from '../util/logger';
+
+//TODO: resource scope?
+import EventEmitter from 'events';
+
 
 var conf = config();
 const MAX_PARALLEL = 4;
 
 class SpatialService {
-  constructor(zookeeper) {
+  constructor(zookeeper, amq, iss) {
     if (!zookeeper) throw new Error("SpatialService needs zookeeper!");
+    if (!amq) throw new Error("SpatialService needs amq!");
+    if (!iss) throw new Error("SpatialService needs iss!");
+
     this._zk = zookeeper;
+    this._iss = iss;
+    logger.info(`Connected and listening to queue: ${conf.amq.inName}`);
+    amq.subscribe(conf.amq.inName, this._onMessage.bind(this));
   }
 
+  _onMessage(message, headers) {
+    const saneMessage = parseAMQMessage(message);
+    logger.info(`Got a message ${JSON.stringify(saneMessage)}`);
+
+    const activity = this._iss.activity(saneMessage);
+    //Send iss a start message
+    activity.onCreate(
+      saneMessage.user,
+      saneMessage.view,
```

Comment:
So core has already created the parent view before it puts the message on the GeoImports queue. Is it possible that Core should actually be sending the activity started message, and geo-import should release an event like "geo processing started" or something like that?